mw-webapp/src/service/AuthService.ts

```diff
@@ -0,0 +1,74 @@
+import {onAuthStateChanged, signInWithPopup, signOut} from "firebase/auth";
+import {doc, getDoc, Timestamp} from "firebase/firestore";
+import {auth, db, provider} from "src/firebase";
+import {UserService} from "src/service/UserService";
+
+/**
+ * Provides methods to interact with the Comments collection
+ */
+export class AuthService {
+
+  /**
+   * Call method for log in
+   */
+  public static async logIn() {
+
+    /**
+     * TODO:
+     * Сheck how it works in browsers: Edge, Safari
+     * Using signInWithRedirect works in Chrome but doesn't work in FireFox and maybe other browsers
+     * Using signInWithPopUp works in FireFox but calls in Chrome and Opera Error:
+     * "Cross-Origin-Opener-Policy policy would block the window.closed call."
+     * Link on discuss about this problem: https://github.com/vercel/next.js/discussions/51135
+     * Resolve solutions:
+     * https://firebase.google.com/docs/auth/web/redirect-best-practices
+     */
+    await signInWithPopup(auth, provider);
+  }
+
+  /**
+   * Call method for logout
+   */
+  public static async logOut() {
+    await signOut(auth);
+  }
+
+  /**
+   * Tracks whether the user is logged in or not
+   */
+  public static async listenAuthStateChange(onLogIn: (userUid: string) => void, onLogOut: (user: null) => void) {
```

Comment:
why not to use object? also check signatures, why should we logout null?

Reply:
Created